pages/03-core/standalone-legend.page.tsx

```diff
@@ -0,0 +1,326 @@
+  { x: 1601008200000, y: 204405 },
+  { x: 1601009100000, y: 190391 },
+  { x: 1601010000000, y: 183570 },
+  { x: 1601010900000, y: 162592 },
+  { x: 1601011800000, y: 148910 },
+  { x: 1601012700000, y: null },
+  { x: 1601013600000, y: 293910 },
+];
+
+const dataA = baseline.map(({ x, y }) => ({ x, y }));
+const dataB = baseline.map(({ x, y }) => ({ x, y: y === null ? null : y + randomInt(-100000, 100000) }));
+const dataC = baseline.map(({ x, y }) => ({ x, y: y === null ? null : y + randomInt(-150000, 50000) }));
+
+const pieSeries: Highcharts.SeriesOptionsType[] = [
+  {
+    name: "Resource Allocation",
+    type: "pie",
+    data: [
+      { name: "CPU Utilization", y: 45.0 },
+      { name: "Memory Usage", y: 26.8 },
+      { name: "Storage Capacity", y: 12.8 },
+    ],
+  },
+];
+
+const lineSeries: Highcharts.SeriesOptionsType[] = [
+  {
+    name: "CPU Utilization",
+    type: "line",
+    data: dataA,
+  },
+  {
+    name: "Memory Usage",
+    type: "line",
+    data: dataB,
+  },
+  {
+    name: "Storage Capacity",
+    type: "line",
+    data: dataC,
+  },
+  {
+    name: "X",
+    type: "scatter",
+    data: [{ x: 1601012700000, y: 500000 }],
+    marker: { symbol: "square" },
+    showInLegend: false,
+  },
+];
+
+const legendItems: readonly LegendItem[] = [
+  {
+    id: "CPU Utilization",
+    name: "CPU Utilization",
+    marker: <ChartSeriesMarker color={colors[0]} type={"square"} visible={true} />,
+    visible: true,
+    highlighted: false,
+  },
+  {
+    id: "Memory Usage",
+    name: "Memory Usage",
+    marker: <ChartSeriesMarker color={colors[1]} type={"square"} visible={true} />,
+    visible: true,
+    highlighted: false,
+  },
+  {
+    id: "Storage Capacity",
+    name: "Storage Capacity",
+    marker: <ChartSeriesMarker color={colors[2]} type={"square"} visible={true} />,
+    visible: true,
+    highlighted: false,
+  },
+];
+
+export default function () {
+  const { chartProps } = useChartSettings();
+
+  const [items, setItems] = useState(legendItems);
+  const pieChartAPI = useRef<CoreChartProps.ChartAPI | null>(null);
+  const lineChartAPI = useRef<CoreChartProps.ChartAPI | null>(null);
+
+  const legendProps = {
+    title: chartProps.cartesian.legend?.title,
+    actions: chartProps.cartesian.legend?.actions,
+  } as const;
+
+  const clearLegendHighlight = () => {
+    setItems((prevItems) => {
+      const hiddenSeries = prevItems.find((i) => !i.visible) !== undefined;
+      return prevItems.map(({ ...i }) => ({ ...i, highlighted: hiddenSeries ? i.visible : false }));
+    });
+  };
+
+  const highlightLegendItem = (itemId: string) => {
+    setItems((prevItems) => {
+      return prevItems.map((i) => ({ ...i, highlighted: i.id === itemId }));
+    });
+  };
+
+  const changeVisibleLegentItems = (itemIds: readonly string[]) => {
+    setItems((prevItems) => {
+      return prevItems.map((i, idx) => {
+        const visible = itemIds.includes(i.id);
+        return {
+          ...i,
```

Comment:
nit: I would call the variables properly: `item`, `index`, as otherwise the code is harder to grasp (the `i` usually stands for `index`, and `idx` can mean `indices` sometimes).